packages/astro/src/default/pages/[...slug].astro

```diff
@@ -17,7 +17,7 @@ type Props = InferGetStaticPropsType<typeof getStaticPaths>;
 const { lesson, logoLink, navList, title } = Astro.props as Props;
 ---
 
-<Layout title={title}>
+<Layout title={title} description="A TutorialKit interactive lesson">
```

Comment:
Ideally this should come from the metadata. You can grab it from `Astro.props` here and then it needs to be returned as part of `generateStaticRoutes`.

Reply:
@Nemikolh I would advise to improve the collection, rather than `generateStaticRoutes`, to allow feeding more data to the "lesson" object. 
Because `generateStaticRoutes` is specific to this page only and is difficult to reuse if you want to craft a sitemap or an RSS feed. And you still need a way to pass some metadata from your MDX file to feed "generateStaticRoutes". That's why I couldn't come up with a more customizable/elegant solution yet.

Reply:
Hey! I think I should have provided more pointers, sorry!

Let me explain. So here's what we need to do:

* Add a `metaTagsSchema` to https://github.com/stackblitz/tutorialkit/blob/main/packages/types/src/schemas/common.ts:

```ts
export const metaTagsSchema = z.object({
  // ideally we would have want to use image from 
  // https://docs.astro.build/en/guides/images/#images-in-content-collections
  // but we can't so we'll need a custom logic (see below)
  image: z.string().optional().describe('An image to show on social previews'),
  description: z.string().optional().describe('...'),
  title: z.string().optional().describe('...'),
})

...

export const webcontainerSchema = commandsSchema.extend({
  meta: metaTagsSchema.optional()
    .describe('Configure Open graph meta tags for previews on social medias'),
  ...
});
```

* Add `'meta'` here:

https://github.com/stackblitz/tutorialkit/blob/f49e9107d35b98079a0fb16c74b9f37a45357661/packages/astro/src/default/utils/content.ts#L238-L240

* Now we could handle `image` in the same file when the collection is traversed to resolve all the relative path to be fully resolved. However that won't be enough to have them included in the final bundle. Instead for now we can limit ourselves to images residing in the `public` folder (so starting with `/`).

* Finally update `[...slug].astro` to use the new `lesson.meta?.<prop>` properties

Reply:
@Nemikolh right thanks for the insight, I was not sure if this should be implemented in user-land or not but it seems a sound idea to have at least a title, description and image default setting, having a finer grained system (eg you want a different image for OpenGraph and Twitter, or images outsides of /public or even generated via an API) would be worth a different PR I think

Reply:
I agree! Yeah, let's start with a simple set of settings and move to a finer grained system later :+1: